internal/ethapi/api.go

```diff
@@ -954,6 +954,8 @@ func DoCall(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash
 		evm.Cancel()
 	}()
 
+	// Call Prepare to clear out the statedb access list
+	state.Prepare(common.Hash{}, 0)
```

Comment:
I checked, and it seems Prepare does not clear the access list.

Reply:
Prepare was changed not to clear the accessList in this PR: https://github.com/ethereum/go-ethereum/pull/24515

Reply:
> 

Sorry I didn't notice this change, my fix is based on an old code.

So now there's no issue on estimate gas, this PR should be rejected.